test/custom_verbose_configuration_tests.cpp

```diff
@@ -0,0 +1,74 @@
+#include "adobe/contract_checks.hpp"
+#include "portable_death_tests.hpp"
+#include <array>
+#include <cstdint>
+#include <cstdio>
+#include <exception>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <stdexcept>
+
+[[noreturn]] void adobe_contract_violated_verbose(const char *condition,
+  adobe::contract_violation_kind kind,
+  const char *file,
+  std::uint32_t line,
+  const char *message)
+{
+  std::array<char, 1024> output{};
+  (void)std::snprintf(output.data(),
+    sizeof(output),
+    "%s:%d: %s (%s). %s\n",
+    file,
+    static_cast<int>(line),
+    kind == adobe::contract_violation_kind::precondition ? "Precondition violated"
+                                                         : "Invariant not upheld",
+    condition,
+    message);
+  throw std::logic_error(output.data());
```

Comment:
I don't understand why these things throw.  If you want good tests for the handler getting called correctly, you don't print anything to stdout; you capture all the data and do the comparisons in C++ code.  Set globals and compare against those if you have to, then exit with success if the test passed and failure otherwise.  Then you don't have to make these death tests at all.

Reply:
They are not actually death tests (fixing the naming). I'm also treating these as a test case for expected use.

Reply:
Then you should do the 2nd thing; no printing to stdout and just test the information passed to the handler by looking at the exception.  If you just look back in the commit history you'll see the the tests I used to have for customized handlers work that way.

Reply:
This isn't printing to stdout - just building a string for `what()`.

Reply:
Oh.  OK, eww.  How about using C++ facilities for this instead of unsafe C ones?

Reply:
Okay.

Reply:
Done.